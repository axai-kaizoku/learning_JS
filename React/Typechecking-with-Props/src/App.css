.app {
  position: absolute;
  width: 445px;
  height: 410px;
  border: 1px solid;
  background-color: rgba(0, 0, 0, 0.1);
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
  margin: auto;
}

.user-card {
  position: relative;
  width: 400px;
  padding: 10px;
  height: 100px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.2);
  margin: 10px;
  border-radius: 5px;
  background-color: white;
}

.user-card > img {
  position: absolute;
  top: 25px;
  left: 20px;
  border-radius: 40px;
}

.user-name {
  position: absolute;
  font-size: 20px;
  left: 120px;
  top: 38px;
  font-weight: bold;
}

.designation {
  position: absolute;
  font-size: 16px;
  left: 120px;
  top: 60px;
  color: rgb(0, 54, 0);
}

.admin {
  font-size: 30px;
  position: absolute;
  right: 20px;
  top: 40px;
  background-color: red;
  color: white;
  padding: 10px;
  border-radius: 20px;
  width: 20px;
  height: 20px;
  text-align: center;
  line-height: 16px;
}
