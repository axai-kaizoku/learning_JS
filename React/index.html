<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="UTF-8" />
		<meta
			name="viewport"
			content="width=device-width, initial-scale=1.0" />
		<title>React</title>
		<style>
			#app {
				border: 1px solid black;
				width: 300px;
				height: 100px;
				margin-top: 200px;
				margin-left: 100px;
			}
		</style>
	</head>
	<body>
		<div id="app"></div>
		<script src="https://unpkg.com/react@16/umd/react.development.js"></script>
		<script src="https://unpkg.com/react-dom@16/umd/react-dom.development.js"></script>
		<script src="https://unpkg.com/@babel/standalone/babel.min.js"></script>

		<script type="text/babel">
			class MyApp extends React.Component {
				constructor() {
					super();
					this.state = {
						txt: '',
					};
				}
				render() {
					return (
						<>
							<div className='top'>
								<input
									type='text'
									id='inputFld'
									onChange={(e) => this.setState({ txt: e.target.value })}
								/>
							</div>
							<div className='bottom'>
								<span className='output'>
									{this.state.txt.split('').reverse().join('')}
								</span>
							</div>
						</>
					);
				}
			}

			ReactDOM.render(<MyApp />, document.getElementById('app'));
		</script>
	</body>
</html>
