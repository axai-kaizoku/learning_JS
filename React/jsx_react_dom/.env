REACT_APP_API_KEY = 7c2d3891c52c48624d436d40a8ba77cb