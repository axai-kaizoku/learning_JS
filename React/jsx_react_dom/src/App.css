.app {
  position: absolute;
  width: 445px;
  height: 730px;
  border: 1px solid;
  background-color: rgba(0, 0, 0, 0.1);
  top: 0px;
  bottom: 0px;
  left: 0px;
  right: 0px;
  margin: auto;
}
.movie-card {
  position: relative;
  height: 50px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  margin: 10px;
  padding: 5px;
  background-color: white;
  border-radius: 10px;
}

.title {
  position: absolute;
  font-size: 20px;
  left: 20px;
  top: 5px;
}

.genre {
  position: absolute;
  left: 20px;
  top: 30px;
  font-size: 12px;
  font-weight: bold;
}

.genre-label {
  position: relative;
  display: inline-block;
  margin-right: 2px;
  background-color: rgb(184, 2, 2);
  color: white;
  padding: 2px;
  padding-right: 5px;
  padding-left: 5px;
  border-radius: 2px;
}
