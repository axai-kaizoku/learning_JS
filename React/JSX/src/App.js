import React from 'react';
import './App.css';
import booksDb from './booksDb';
import Book from './Book';

const App = () => {
	return (
		<div className='app-container'>
			<div className='app-title'>Great Books</div>
			{booksDb.map((book) => (
				<Book
					title={book.title}
					author={book.author}
					key={book.id}
				/>
			))}
		</div>
	);
};

export default App;
