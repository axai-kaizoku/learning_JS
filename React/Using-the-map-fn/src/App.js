import React from 'react';
// import Users from './components/Users';
import Inbox from './components/Inbox';
{
	/* <Users/> */
}
const App = () => (
	<div className="main">
		<Inbox />
	</div>
);

export default App;
