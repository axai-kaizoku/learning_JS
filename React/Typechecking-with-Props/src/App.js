import React from "react";
import "./App.css";
import users from "./userDb";
import UserCard from "./UserCard";

const App = () => {
  return (
    <div className="app">
      {users.map(({ id, name, designation, isAdmin, photo }) => (
        <UserCard
          key={id}
          name={name}
          designation={designation}
          isAdmin={isAdmin}
          photo={photo}
        />
      ))}
    </div>
  );
};

export default App;
