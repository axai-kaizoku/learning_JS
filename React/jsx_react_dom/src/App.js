import React from 'react';
import './App.css';
import MovieCard from './MovieCard';
import moviesDb from './moviesDb';

const App = () => {
	return (
		<div className="app">
			{moviesDb.map(({ movie_title, movie_genre }) => (
				<MovieCard
					title={movie_title}
					genre={movie_genre}
				/>
			))}
		</div>
	);
};

export default App;
