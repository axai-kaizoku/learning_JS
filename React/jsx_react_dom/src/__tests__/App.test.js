import React from 'react';
import { render, fireEvent, cleanup, wait } from '@testing-library/react';
import App from '../App';

jest.mock('../services', () => {
	return {
		getToDos: () =>
			Promise.resolve([
				{
					title: 'Book concert tickets',
					done: false,
					id: 1,
				},
				{
					title: 'Buy a new iPhone',
					done: true,
					id: 2,
				},
				{
					title: 'Write an article on React',
					done: true,
					id: 3,
				},
			]),
		addTodo: (obj) => Promise.resolve({ ...obj, id: 4 }),
	};
});

afterEach(cleanup);

describe('Snapshot testing the App', () => {
	it('Renders a Snapshot successfully', async () => {
		const { asFragment, getByPlaceholderText } = render(<App />);
		await wait(() => expect(asFragment()).toMatchSnapshot());

		const taskInput = getByPlaceholderText(/Add a task/i);
		fireEvent.change(taskInput, {
			target: { value: 'Write some integration tests for the project' },
		});
		fireEvent.keyUp(taskInput, { key: 'Enter', keyCode: 13 });
		await wait(() => expect(asFragment()).toMatchSnapshot());
	});
});
