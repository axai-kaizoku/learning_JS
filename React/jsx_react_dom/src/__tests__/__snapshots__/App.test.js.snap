// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Using map to render lists List of movies render as expected! 1`] = `"<div class=\\"app\\"><div class=\\"movie-card\\"><div class=\\"title\\">The Hanoi Hilton</div><div class=\\"genre\\"><div class=\\"genre-label\\">Drama</div><div class=\\"genre-label\\">War</div></div></div><div class=\\"movie-card\\"><div class=\\"title\\">American Teen</div><div class=\\"genre\\"><div class=\\"genre-label\\">Documentary</div></div></div><div class=\\"movie-card\\"><div class=\\"title\\">Monster, The (Mostro, Il)</div><div class=\\"genre\\"><div class=\\"genre-label\\">Comedy</div></div></div><div class=\\"movie-card\\"><div class=\\"title\\">What Have I Done to Deserve This?</div><div class=\\"genre\\"><div class=\\"genre-label\\">Comedy</div><div class=\\"genre-label\\">Drama</div></div></div><div class=\\"movie-card\\"><div class=\\"title\\">World on a Wire</div><div class=\\"genre\\"><div class=\\"genre-label\\">Crime</div><div class=\\"genre-label\\">Sci-Fi</div></div></div><div class=\\"movie-card\\"><div class=\\"title\\">Bad Boy Bubby</div><div class=\\"genre\\"><div class=\\"genre-label\\">Drama</div></div></div><div class=\\"movie-card\\"><div class=\\"title\\">Cuenca Crime</div><div class=\\"genre\\"><div class=\\"genre-label\\">Drama</div></div></div><div class=\\"movie-card\\"><div class=\\"title\\">Blue Sunshine</div><div class=\\"genre\\"><div class=\\"genre-label\\">Horror</div><div class=\\"genre-label\\">Thriller</div></div></div><div class=\\"movie-card\\"><div class=\\"title\\">Men with Brooms</div><div class=\\"genre\\"><div class=\\"genre-label\\">Comedy</div><div class=\\"genre-label\\">Drama</div><div class=\\"genre-label\\">Romance</div></div></div><div class=\\"movie-card\\"><div class=\\"title\\">Avenging Angelo</div><div class=\\"genre\\"><div class=\\"genre-label\\">Action</div><div class=\\"genre-label\\">Comedy</div><div class=\\"genre-label\\">Crime</div></div></div></div>"`;
